src/ray/object_manager/object_manager.cc

```diff
@@ -707,28 +707,32 @@ void ObjectManager::ProcessNewClient(TcpClientConnection &conn) {
 
 void ObjectManager::ProcessClientMessage(std::shared_ptr<TcpClientConnection> &conn,
                                          int64_t message_type, const uint8_t *message) {
-  switch (message_type) {
-  case static_cast<int64_t>(object_manager_protocol::MessageType::PushRequest): {
+  if (static_cast<protocol::MessageType>(message_type) ==
+      protocol::MessageType::DisconnectClient) {
```

Comment:
It seems problematic that `message_type` can have type either `protocol::MessageType` or `object_manager_protocol::MessageType` because those two enums could have overlapping values. For example, couldn't `protocol::MessageType::DisconnectClient` be the same integer value as some value in `object_manager_protocol::MessageType`?

Reply:
That's right. Currently `object_manager_protocol::MessageType` doesn't have the same value as `protocol::MessageType::DisconnectClient`. But we may accidentally introduce bugs in the future when we add new values to `object_manager_protocol::MessageType`.

I don't know what's the reason why this `message_type` uses mixed types. Can we also add a `DisconnectClient` to `object_manager_protocol::MessageType`? @zhijunfu 

If that cannot be easily fixed, we should at least add a warning comment in `object_manager.fbs`

Reply:
Yes this is indeed a problem, although it's not brought in by this PR. 
How about we support both `protocol::MessageType::DisconnectClient` and `object_manager_protocol::MessageType::DisconnectClient`? Then two different sets of `MessageType` enums can be totally decoupled.

Reply:
yeah, that would be nicer.